src/intervals/rounding.jl

```diff
@@ -4,53 +4,61 @@ This is a so-called "traits-based" design, as follows.
-+(IntervalRounding{:fast}, a, b, RoundDown)
-+(IntervalRounding{:tight}, a, b, RoundDown)
-+(IntervalRounding{:accurate}, a, b, RoundDown)
-+(IntervalRounding{:slow}, a, b, RoundDown)
-+(IntervalRounding{:none}, a, b, RoundDown)
++(IntervalRounding{:fast}(), a, b, RoundDown)
++(IntervalRounding{:tight}(), a, b, RoundDown)
++(IntervalRounding{:accurate}(), a, b, RoundDown)
++(IntervalRounding{:slow}(), a, b, RoundDown)
++(IntervalRounding{:none}(), a, b, RoundDown)
 
 The current allowed rounding types are
-- :fast     # fast, tight (correct) rounding with errorfree arithmetic via FastRounding.jl
-- :tight # tight (correct) rounding with improved errorfree arithmetic via RoundingEmulator.jl
-- :accurate # fast "accurate" rounding using prevfloat and nextfloat  (slightly wider than needed)
-- :slow    # tight (correct) rounding by changing rounding mode (slow)
-- :none     # no rounding (for speed comparisons; no enclosure is guaranteed)
-
-The function `setrounding(Interval, rounding_type)` then defines rounded
- functions *without* an explicit rounding type, e.g.
-
-sin(x, r::RoundingMode) = sin(IntervalRounding{:slow}, x, r)
-
-These are overwritten when `setrounding(Interval, rounding_type)` is called again.
-
-In Julia v0.6 and later (but *not* in Julia v0.5), this automatically redefines all relevant functions, in particular those used in +(a::Interval, b::Interval) etc., so that all interval functions automatically use the correct interval rounding type from then on.
+- :fast  # fast, tight (correct) rounding with errorfree arithmetic via FastRounding.jl
+- :tight  # tight (correct) rounding with improved errorfree arithmetic via RoundingEmulator.jl
+- :accurate  # fast "accurate" rounding using prevfloat and nextfloat  (slightly wider than needed)
+- :slow  # tight (correct) rounding by changing rounding mode (slow)
+- :none  # no rounding (for speed comparisons; no enclosure is guaranteed)
+
+All function on intervals are then defined by default as
+    +(IntervalArithmetic.interval_rounding(), a, b, RoundDown)
+
+By rededfining `IntervalArithmetic.interval_rounding()`, the user can
+select the default.
+
+Moreover the `:slow` rounding mode is used as a fallback if the chosen
+rounding mode is not available.
+This is done by defining the generic case as
+    +(::IntervalRounding, a, b, RoundDown) = +(IntervalRounding{:slow}(), a, b, RoundDown)
 =#
 
 
-"""Interval rounding trait type"""
-struct IntervalRounding{T} end
-
-
+"""
+    IntervalRounding{T}
 
+Interval rounding trait type.
 
-# Functions that are the same for all rounding types:
+Allowed rounding types are
+- `:tight`: fast, tight (correct) rounding with errorfree arithmetic via
+            FastRounding.jl.
+- `:accurate`: fast "accurate" rounding using prevfloat and nextfloat
+               (slightly wider than needed).
+- `:slow`: tight (correct) rounding by changing rounding mode (slow).
+- `:none`: no rounding (for speed comparisons; no enclosure is guaranteed).
```

Comment:
The `:fast` mode is missing in this description.

Reply:
also I think `:tight` should be using `RoundingEmulator.jl` ?